cpp/include/cuml/fil/fil.h

```diff
@@ -243,9 +244,10 @@ struct treelite_params_t {
   float threshold;
   // storage_type indicates whether the forest should be imported as dense or sparse
   storage_type_t storage_type;
-  // blocks_per_sm, if set, works as a limit to improve cache hit rate for larger forests
+  // blocks_per_sm, if nonzero, works as a limit to improve cache hit rate for larger forests
   // suggested values (if nonzero) are from 2 to 7
-  int blocks_per_sm = 0;
+  // if zero, launches ceildiv(num_rows, NITEMS) blocks
+  int blocks_per_sm;
```

Comment:
`'auto'` can map to `0`, which indeed determines the number of thread blocks per SM based on occupancy. I will make this change.

Reply:
I thought 0 determines the number of thread blocks based on the number of rows. In any case, I'd reserve `'auto'` for something different (e.g. determine the number of thread blocks per SM based on some heuristic), and have explicit names in Python for other modes.

Reply:
I am wondering if we should make that the default mode because it will never be slower than the current default (since __syncthreads() at the end is unconditional and does not work with wildly divergent threads)

Reply:
a quick measurement showed speedups and slowdowns on the order of a couple %, so there isn't a strong motivation to change it yet